pkg/k8shandler/cluster.go

```diff
@@ -125,13 +133,15 @@ func (elasticsearchRequest *ElasticsearchRequest) CreateOrUpdateElasticsearchClu
 					}
 
 					addNodeState(node, nodeStatus)
-					elasticsearchRequest.setNodeStatus(node, nodeStatus, clusterStatus)
+					if err := elasticsearchRequest.setNodeStatus(node, nodeStatus, clusterStatus); err != nil {
+						return err
+					}
 
 					elasticsearchRequest.updateMinMasters()
 				}
 
 				// we only want to update our replicas if we aren't in the middle up an upgrade
-				UpdateReplicaCount(
+				return UpdateReplicaCount(
```

Comment:
we don't want to return here because we want to return from `UpdateClusterStatus`

Reply:
Fixed by printing the error instead of returning it.